node/blockAPI/shardBlock_test.go

```diff
@@ -0,0 +1,304 @@
+
+	"github.com/ElrondNetwork/elrond-go/data/api"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/node/mock"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/testscommon"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestShardAPIBlockProcessor_GetBlockByHash_InvalidHashShouldErr(t *testing.T) {
+	t.Parallel()
+
+	shardID := uint32(3)
+	headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+
+	storerMock := mock.NewStorerMock()
+	uint64Converter := mock.NewNonceHashConverterMock()
+	shardAPIBlockProcessor := NewShardApiBlockProcessor(
+		&APIBlockProcessorArg{
+			SelfShardID: shardID,
+			Marshalizer: &mock.MarshalizerFake{},
+			Store: &mock.ChainStorerMock{
+				GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+					return storerMock
+				},
+				GetCalled: func(unitType dataRetriever.UnitType, key []byte) ([]byte, error) {
+					return headerHash, nil
+				},
+			},
+			Uint64ByteSliceConverter: uint64Converter,
+			HistoryRepo: &testscommon.HistoryRepositoryStub{
+				IsEnabledCalled: func() bool {
+					return true
+				},
+				GetEpochByHashCalled: func(hash []byte) (uint32, error) {
+					return 1, nil
+				},
+			},
+		},
+	)
+
+	blk, err := shardAPIBlockProcessor.GetBlockByHash([]byte("invalidHash"), false)
+	assert.Nil(t, blk)
+	assert.Error(t, err)
+}
+
+func TestShardAPIBlockProcessor_GetBlockByNonce_InvalidNonceShouldErr(t *testing.T) {
+	t.Parallel()
+
+	shardID := uint32(3)
+	headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+
+	storerMock := mock.NewStorerMock()
+	uint64Converter := mock.NewNonceHashConverterMock()
+	shardAPIBlockProcessor := NewShardApiBlockProcessor(
+		&APIBlockProcessorArg{
+			SelfShardID: shardID,
+			Marshalizer: &mock.MarshalizerFake{},
+			Store: &mock.ChainStorerMock{
+				GetStorerCalled: func(unitType dataRetriever.UnitType) storage.Storer {
+					return storerMock
+				},
+				GetCalled: func(unitType dataRetriever.UnitType, key []byte) ([]byte, error) {
+					return headerHash, nil
+				},
+			},
+			Uint64ByteSliceConverter: uint64Converter,
+			HistoryRepo: &testscommon.HistoryRepositoryStub{
+				IsEnabledCalled: func() bool {
+					return false
+				},
+			},
+		},
+	)
+
+	blk, err := shardAPIBlockProcessor.GetBlockByNonce(100, false)
+	assert.Nil(t, blk)
+	assert.Error(t, err)
+}
+
+func TestShardAPIBlockProcessor_GetBlockByHashFromNormalNode(t *testing.T) {
+	t.Parallel()
+
+	nonce := uint64(1)
+	round := uint64(2)
+	epoch := uint32(1)
+	shardID := uint32(3)
+	miniblockHeader := []byte("miniBlockHash")
+	headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+
+	storerMock := mock.NewStorerMock()
+	uint64Converter := mock.NewNonceHashConverterMock()
+	shardAPIBlockProcessor := NewShardApiBlockProcessor(
+		&APIBlockProcessorArg{
```

Comment:
This looks like repeated, duplicated code. Can we have a function that generates this argument? (maybe the function should have the parameters required by all use cases?) Valid for the other test files.

Reply:
added createMockArgs function for each test file